Source/Core/Core/HW/WII_IPC.h

```diff
@@ -35,6 +36,71 @@ enum StarletInterruptCause
   INT_CAUSE_IPC_STARLET = 0x80000000
 };
 
+enum class GPIO : u32
+{
+  POWER = 0x1,
+  SHUTDOWN = 0x2,
+  FAN = 0x4,
+  DC_DC = 0x8,
+  DI_SPIN = 0x10,
+  SLOT_LED = 0x20,
+  EJECT_BTN = 0x40,
+  SLOT_IN = 0x80,
+  SENSOR_BAR = 0x100,
+  DO_EJECT = 0x200,
+  EEP_CS = 0x400,
+  EEP_CLK = 0x800,
+  EEP_MOSI = 0x1000,
+  EEP_MISO = 0x2000,
+  AVE_SCL = 0x4000,
+  AVE_SDA = 0x8000,
+  DEBUG0 = 0x10000,
+  DEBUG1 = 0x20000,
+  DEBUG2 = 0x40000,
+  DEBUG3 = 0x80000,
+  DEBUG4 = 0x100000,
+  DEBUG5 = 0x200000,
+  DEBUG6 = 0x400000,
+  DEBUG7 = 0x800000,
+};
+
+class GPIOBit
+{
+public:
+  GPIOBit(u32& bits, GPIO bit) : m_bits(bits), m_bit(bit) {}
+  explicit operator bool() { return (m_bits & static_cast<u32>(m_bit)) != 0; }
```

Comment:
I missed this in my initial review, but this should likely also have a const specifier like so:

```suggestion
  explicit operator bool() const { return (m_bits & static_cast<u32>(m_bit)) != 0; }
```

Reply:
Ah, that's where that should go; I had `explicit operator bool const()` but GCC seemed to dislike it because `const bool` doesn't really mean much for a return type.